### PR TITLE
feat(telemetry): allow to disable telemetry through settings INTELLIJ-12

### PR DESCRIPTION
**Pull request by kmruiz**
## Description

Notifies a plugin user, the first time the plugin is activated, that we use anonymous telemetry. In the notification balloon there is an action to disable the telemetry from there. In the plugin settings there is a checkbox that allows us to toggle telemetry any time.

signal-2024-06-12-14-43-50-481
signal-2024-06-12-14-43-50-481-1

### Checklist

- [x] New tests and/or benchmarks are included.
- [ ] Documentation is changed or added.
- [x] Changelog is updated accordingly.
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement).

### CODE REVIEW

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/ActivatePluginPostStartupActivity.kt [hunk 1]
```diff
@@ -1,13 +1,34 @@
+/**
+ * These class implement the balloon that shows the first time that the plugin is activated.
```
Comment (addaleax) [inline packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/ActivatePluginPostStartupActivity.kt:2]: typo?

```suggestion
 * These classes implement the balloon that shows the first time that the plugin is activated.
```

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/settings/PluginSettings.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/**
+ * Settings state for the plugin. These classes are responsible for the persistence of the plugin
+ * settings.
```
Comment (addaleax) [inline packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/settings/PluginSettings.kt:3]: How "broad" is the scope of settings in IntelliJ? Is every setting defined here automatically visible and configurable to the user, or is it more like Compass, where it's fine for us to use some settings as easy ways of persisting small bits of data?

File: packages/jetbrains-plugin/src/main/resources/messages/TelemetryBundle.properties [hunk 3]
```diff
@@ -0,0 +1,6 @@
+notification.group.name=MongoDB telemetry
+notification.title=MongoDB plugin telemetry
+notification.message=Anonymous telemetry is enabled by default because it helps us improve the plugin.
+action.disable-telemetry=Disable Telemetry
+settings.telemetry-collection-tooltip=Allow the collection of anonymous diagnostics and usage telemetry data to help improve the product.
+settings.telemetry-collection-checkbox=Enable telemetry
```
Comment (addaleax) [inline packages/jetbrains-plugin/src/main/resources/messages/TelemetryBundle.properties:6]: Do we need to/can we include links to https://www.mongodb.com/legal/privacy/privacy-policy?

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/settings/PluginSettings.kt (hunk 2 above)
Comment (kmruiz) [reply packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/settings/PluginSettings.kt:3]: It's more like Compass. So you have the "settings state" that is persisted and it's not related anyhow to the settings view. How you want to handle these settings is up to you. There is a class (com.mongodb.jbplugin.settings.PluginSettingsConfigurable) that specifies how this settings can be configured in the Settings UI and it's not mandatory to add all of them, only whichever are relevant. 

To give you an example, `hasTelemetryOptOutputNotificationBeenShown` is not set up in the Settings page, but is set up in the initial activity (com.mongodb.jbplugin.ActivatePluginPostStartupActivity).

File: packages/jetbrains-plugin/src/main/resources/messages/TelemetryBundle.properties (hunk 3 above)
Comment (kmruiz) [reply packages/jetbrains-plugin/src/main/resources/messages/TelemetryBundle.properties:6]: That's a good point. I'm going to add a link on the notification and on the settings page to the privacy policy.

**Review by addaleax — APPROVED** [review]

**Pull request merged**